framework/behaviors/AttributeBehavior.php

```diff
@@ -85,6 +85,11 @@ class AttributeBehavior extends Behavior
      * @since 2.0.8
      */
     public $skipUpdateOnClean = true;
+    /**
+     * @var bool whether to preserve non-empty attribute values.
+     * @since 2.0.12
+     */
+    public $preserveNonEmptyValues = false;
```

Comment:
skip + not is a double negation which is not ideal. Positive formulation is better for readability. What about `keepSetValues`